graphql/resolvers.go

```diff
@@ -1819,6 +1819,71 @@ func (r *mutationResolver) UnschedulePatchTasks(ctx context.Context, patchID str
 	return &patchID, nil
 }
 
+// ScheduleUndispatchedBaseTasks only allows scheduling undispatched base tasks for tasks that are failing on the current patch
+func (r *mutationResolver) ScheduleUndispatchedBaseTasks(ctx context.Context, patchID string) ([]*restModel.APITask, error) {
+	opts := data.TaskFilterOptions{
+		Statuses:              evergreen.TaskFailureStatuses,
+		IncludeExecutionTasks: true,
+	}
+	tasks, _, err := r.sc.FindTasksByVersion(patchID, opts)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch: %s ", err.Error()))
+	}
+
+	scheduledTasks := []*restModel.APITask{}
+	tasksToSchedule := make(map[string]*task.Task)
+
+	generatedTasksToSchedule := []task.Task{}
+
+	for _, t := range tasks {
+		// If a task is a generated task don't schedule it until we get all of the generated tasks we want to generate
+		if t.GeneratedBy == "" {
+			// We can ignore an error while fetching tasks because this could just mean the task didn't exist on the base commit.
+			baseTask, _ := t.FindTaskOnBaseCommit()
+			if baseTask != nil && baseTask.Status == evergreen.TaskUndispatched {
+				tasksToSchedule[baseTask.Id] = baseTask
+			}
+			// If a task is generated lets keep track of it so we can restart it later
+		} else if t.GeneratedBy != "" {
+			generatedTasksToSchedule = append(generatedTasksToSchedule, t)
+
+		}
+	}
+
+	// If a generated task does not have a base task find its generator and use its base task
+	// to generate it
+	for _, t := range generatedTasksToSchedule {
```

Comment:
I'm not convinced we need this loop; can't we just do this within the else if statement at 1847? Unless you're keeping it broken up for clarity